pkg/controller/controller.go

```diff
@@ -1083,9 +1086,18 @@ func updatedStatefulSet(
 	expected.Status = actual.DeepCopy().Status
 
 	copyAnnotations(expected, actual)
+	updateSpecTemplate(expected)
 	return expected, true, nil
 }
 
+// updates spec template so that rollout update is always performed.
+func updateSpecTemplate(sts *appsv1.StatefulSet) {
+	if sts.Spec.Template.Annotations == nil {
+		sts.Spec.Template.Annotations = map[string]string{}
+	}
+	sts.Spec.Template.Annotations[annotations.Rollout] = time.Now().Format(rolloutTimeFormat)
```

Comment:
Nit: can you use the k8s-native [time helpers](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Now)? That should handle formatting for you.

Reply:
Will do. I assume k8s annotation values don't have similar restrictions as label values?